### PR TITLE
handle external cluster status

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: handle external cluster status

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3950

```release-note
NONE
```

### CODE REVIEW

File: pkg/api/v2/types.go [hunk 1]
```diff
@@ -376,6 +376,13 @@ const (
 
 	// DELETING state indicates the cluster is being deleted.
 	DELETING ExternalClusterState = "DELETING"
+
+	// UNSPECIFIED Not set.
+	UNSPECIFIED ExternalClusterState = "UNSPECIFIED"
```
Comment (maciaszczykm) [inline pkg/api/v2/types.go]: In other places we use "unknown". What do you think about aligning it?

Comment (zreigz) [reply pkg/api/v2/types.go]: sounds good, I will change

Comment (zreigz) [reply pkg/api/v2/types.go]: done

**Pull request merged**